js/client-chat.js

```diff
@@ -1455,15 +1455,17 @@
 			var lastMessageDates = Tools.prefs('logtimes') || (Tools.prefs('logtimes', {}), Tools.prefs('logtimes'));
 			if (!lastMessageDates[Config.server.id]) lastMessageDates[Config.server.id] = {};
 			var lastMessageDate = lastMessageDates[Config.server.id][this.id] || 0;
-			var mayNotify = msgTime > lastMessageDate && userid !== app.user.get('userid');
+			// because the time offset to the server can vary slightly, subtract it to not have it affect comparisons between dates
+			var serverMsgTime = msgTime - ~~(this.timeOffset);
```

Comment:
It was used twice elsewhere in client-chat (L1171 and 1181), and I figured it was a quick way of making sure that nothing breaks when this.timeOffset is undefined. Can it be assumed that it'll always be defined (i.e. that the server will always send the time offset message first)?